core/types/block.go

```diff
@@ -273,6 +273,7 @@ func (b *Block) GasLimit() uint64     { return b.header.GasLimit }
 func (b *Block) GasUsed() uint64      { return b.header.GasUsed }
 func (b *Block) Difficulty() *big.Int { return new(big.Int).Set(b.header.Difficulty) }
 func (b *Block) Time() uint64         { return b.header.Time }
+func (b *Block) Timestamp() *big.Int  { return new(big.Int).SetUint64(b.header.Time) }
```

Comment:
We should also switch to passing in the header to `CheckConfigurePrecompiles` so that we don't need to use `types.NewBlockWithHeader(head)` in `miner/worker.go` and it contains all of the information needed for this function.

Reply:
we cannot use header because it causes cyclic import.

Reply:
I think it's also cleaner to use a function returning bigInt for the `IsForkTransition`. In order to implement the `BlockContext`  with `Time()` we need to implement it in the https://github.com/ava-labs/subnet-evm/blob/fee-config-precompile/core/vm/evm.go#L119 which also conflicts with the field name `Time`. It's like an rabbit hole there :D